cmd/sonobuoy/app/gen.go

```diff
@@ -22,21 +22,50 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"k8s.io/client-go/kubernetes"
 
 	ops "github.com/heptio/sonobuoy/pkg/client"
 	"github.com/heptio/sonobuoy/pkg/errlog"
 )
 
-var genopts ops.GenConfig
-
-var genFlags struct {
+type genFlags struct {
 	sonobuoyConfig SonobuoyConfig
 	mode           ops.Mode
 	rbacMode       RBACMode
 	kubecfg        Kubeconfig
 }
 
+var genopts ops.GenConfig
+var genflags genFlags
+
+func (g *genFlags) AddFlags(flags *pflag.FlagSet, cfg *ops.GenConfig, rbac RBACMode) {
```

Comment:
What do you think about a slight restructure where this returned a flagset? Then we could have something like:

```
cmd.Flags().AddFlagSet(GenFlagSet(&genopts))
```

This could give us a nice set of functions (GenFlagSet, RunFlagSet, etc) that returned configured flagsets for each command (and also be able to build on top of each other).

RunFlagSet might be like
```
runFlagSet := new(pflag.FlagSet)
runFlagSet.AddFlagSet(GenFlagSet(&genOptions))
runFlagSet.AddFlag(runflag1 ...)
runFlagSet.AddFlag(runflag2 ...)
return runFlagSet
```

I think the the solution you've provided is a solid way to go and a huge improvement, but I find it a little difficult to follow mutating pointers around. I've found it's best to avoid pointers as arguments if it can be avoided.

Reply:
I can give that a try, I think that looks good!